automation/SetUp.ps1

```diff
@@ -26,87 +29,135 @@ if ([string]::IsNullOrEmpty($KubernetesVersion)) {
 }
 
 
-"* Starting the $VMName Virtual Machine ..." > logs
+"{0} - * Starting the $VMName Virtual Machine ..." -f (Get-Date) > logs
 Write-Output "* Starting the $VMName Virtual Machine ..."
 
-$VM = @{
-    Name = $VMName;
-    Generation = 1;
-    MemoryStartupBytes = 1GB;
-    NewVHDPath = "${env:homepath}\.minikube\machines\$VMName\VHD.vhdx";
-    NewVHDSizeBytes = 15GB;
-    BootDevice = "VHD";
-    Path = "${env:homepath}\.minikube\machines\";
-    SwitchName = $SwitchName
-}
-
-Write-Output "* Please wait as we set up the $VMName Virtual Machine ..."
-New-VM @VM | Out-Null
-Set-VM -Name $VMName -ProcessorCount 2 -AutomaticCheckpointsEnabled $false
-Set-VMProcessor -VMName $VMName -ExposeVirtualizationExtensions $true
-Set-VMDvdDrive -VMName $VMName -Path $ISOFilePath
-# Add-VMDvdDrive -VMName $VMName -Path "$PSScriptRoot\auto-install.iso" -ControllerNumber 1 -ControllerLocation 1
-Start-VM -Name $VMName | Out-Null
+if($Platform -eq "Hyper-V"){
+    $VM = @{
```

Comment:
Split HyperV and Virtual box to seperate functions to make it easier to mange each of them

Reply:
yup wlil do